docs/index.js

```diff
@@ -0,0 +1,621 @@
+      url: "wss://ws.tn10.kaspa.forbole.com/borsh",
+      networkId: "testnet-10",
+    });
+    await rpc.connect();
+
+    try {
+      const sendP2SHAddress =
+        document.getElementById("sendP2SHAddress").innerText;
+      const sendScriptBuilder = kaspaWasm.ScriptBuilder.fromScript(
+        document.getElementById("sendScript").innerText,
+      );
+      const sellerAddress = document.getElementById("address").innerText;
+
+      let sendP2SHEntries = [];
+      while (sendP2SHEntries.length === 0) {
+        const sendP2SHUTXOs = await rpc.getUtxosByAddresses([
+          sendP2SHAddress.toString(),
+        ]);
+        sendP2SHEntries = sendP2SHUTXOs.entries;
+      }
+
+      const tx = new kaspaWasm.createTransaction(
+        [sendP2SHEntries[0]],
+        [{ address: sellerAddress, amount: kaspaWasm.kaspaToSompi("1") }],
+        0n,
+      );
+
+      const preparedTxJson = tx.serializeToSafeJSON();
+      const signedTx = await kastle.signTx("testnet-10", preparedTxJson, [
+        {
+          inputIndex: 0,
+          scriptHex: sendScriptBuilder.toString(),
+          signType: "SingleAnyOneCanPay",
+        },
+      ]);
+      document.getElementById("sendPreparedTx").innerText = signedTx;
+      document.getElementById("sendPreparedError").innerText = "None";
+    } catch (error) {
+      document.getElementById("sendPreparedError").innerText = error;
+    } finally {
+      rpc.disconnect();
+    }
+  });
+
+document
+  .getElementById("krcCancelReveal")
+  .addEventListener("click", async () => {
+    const rpc = new kaspaWasm.RpcClient({
+      url: "wss://ws.tn10.kaspa.forbole.com/borsh",
+      networkId: "testnet-10",
+    });
+    await rpc.connect();
+
+    try {
+      const P2SHAddress = document.getElementById("sendP2SHAddress").innerText;
+      const sendScriptBuilder = kaspaWasm.ScriptBuilder.fromScript(
+        document.getElementById("sendScript").innerText,
+      );
+
+      let P2SHEntries = [];
+      while (P2SHEntries.length === 0) {
+        const P2SHUTXOs = await rpc.getUtxosByAddresses([
+          P2SHAddress.toString(),
+        ]);
+        P2SHEntries = P2SHUTXOs.entries;
+
+        if (P2SHEntries.length === 0) {
+          await new Promise((resolve) => setTimeout(resolve, 1000));
+        }
+      }
+
+      const P2SHEntry = P2SHEntries[0];
+      const entry = {
+        address: P2SHEntry.address.toString(),
+        amount: kaspaWasm.sompiToKaspaString(P2SHEntry.amount),
+        scriptPublicKey: JSON.parse(P2SHEntry.scriptPublicKey.toString()),
+        blockDaaScore: P2SHEntry.blockDaaScore.toString(),
+        outpoint: JSON.parse(P2SHEntry.outpoint.toString()),
+      };
+      const revealTxId = await kastle.signAndBroadcastTx("testnet-10", [], {
+        priorityEntries: [entry],
+        scripts: [
+          {
+            scriptHex: sendScriptBuilder.toString(),
+            inputIndex: 0,
+          },
+        ],
+        priorityFee: "0.02",
+      });
```

Comment:
Seller can cancel the list by revealing the send commit script without any new outputs.